satpy/tests/reader_tests/test_nwcsaf_nc.py

```diff
@@ -43,6 +43,20 @@ def setUp(self, xr_, unzip):
         unzip.return_value = ''
         self.scn = NcNWCSAF('filename', {}, {})
 
+    def test_sensor_name(self):
+        """Test that the correct sensor name is being set"""
+
+        self.scn.set_platform_and_sensor(platform_name='Metop-B')
+        self.assertTrue(self.scn.sensors, set(['avhrr-3']))
+        self.scn.set_platform_and_sensor(platform_name='NOAA-20')
+        self.assertTrue(self.scn.sensors, set(['viirs']))
+        self.scn.set_platform_and_sensor(platform_name='Himawari-8')
+        self.assertTrue(self.scn.sensors, set(['ahi']))
+        self.scn.set_platform_and_sensor(platform_name='GOES-17')
+        self.assertTrue(self.scn.sensors, set(['abi']))
+        self.scn.set_platform_and_sensor(platform_name='Meteosat-11')
+        self.assertTrue(self.scn.sensors, set(['seviri']))
```

Comment:
How about checking `sensor_names` instead, so that we are checking the public API, not the filehandler's internals ?

Reply:
Yes, I realized that later, it is added now